src/elements/common/droppable/makeDroppable.js

```diff
@@ -62,22 +62,37 @@ const makeDroppable = ({ dropValidator, onDrop }: { dropValidator?: Function, on
         }
 
         componentDidUpdate() {
-            if (this.droppableEl) {
-                return;
-            }
-
             this.bindDragDropHandlers();
         }
 
         /**
-         * Bind drag and drop event handlers to the droppableEl
+         * Function that removes the drag and drop related event listeners on the input element
+         *
+         * @param {Element} element
+         * @return {void}
+         */
+        removeEventListeners = (element: Element) => {
+            element.removeEventListener('dragenter', this.handleDragEnter);
+            element.removeEventListener('dragover', this.handleDragOver);
+            element.removeEventListener('dragleave', this.handleDragLeave);
+            element.removeEventListener('drop', this.handleDrop);
+        };
+
+        /**
+         * Bind drag and drop event handlers to the droppableEl, when the wrapped element
+         * is changed, remove the event listeners on the previous droppableEl and add
+         * event listeners on the new droppableEl
          */
         bindDragDropHandlers = () => {
             const droppableEl = findDOMNode(this); // eslint-disable-line react/no-find-dom-node
-            if (!droppableEl || !(droppableEl instanceof Element)) {
+            if (!droppableEl || !(droppableEl instanceof Element) || droppableEl === this.droppableEl) {
                 return;
             }
 
+            if (this.droppableEl) {
```

Comment:
Can we move this to `componentDidUpdate`, since it's only relevant for that lifecycle method?

Reply:
Sure. makes sense